docs/DeveloperGuide.md

```diff
@@ -2,37 +2,207 @@
+
+The bulk of the app's work is done by the following components:
+- `UI`: The user interface of the App. It handles user interaction, namely reading user input and displaying messages.
+- `Storage`: Reads data from, and writes data to, the hard disk. 
+Responsible for loading and saving both recipes and ingredients.
+- `Parser`: The command parser. It interprets user inputs,
+turning them into `Command` objects that are executed by the app. 
+- `Command`: Represents actions that the user can perform, such as adding, deleting or listing recipes and ingredients.
+- `RecipeList` and `IngredientList`: These manage collections of recipes and ingredients, respectively. 
+
+
+### How the architecture components interact with each other
+The sequence diagram below shows how the components interact with each other when the application is first launched.
+
+![ApplicationStartUp.png](img/SequenceDiagrams/ApplicationStartUp.png)
+
+(Note: The lifelines of objects cannot be hidden after the X in the PlantUML software used to make this class diagram)
+
+
+### Ui Class
+The Ui class deals with taking in inputs from an input stream (in this case, the user input through `System.in`), 
+and then printing messages read-able by the user (through `System.out`). 
+
+### Recipe Class
+The Recipe class represents a singular recipe added by the user or loaded from the storage database. 
+A valid recipe object contains the following fields:
++ Name
++ List of Ingredients
++ List of Steps
++ Cuisine (optional)
++ Time taken (optional)
+
+The formatting of how a recipe is displayed and saved is handled by this class.
+
+### RecipeList Class
+The RecipeList class represents a list of any non-negative integer number of recipes. 
+This class handles the addition, deletion and editing of recipes.
+
+Shown below is a class diagram detailing the interaction between `RecipeList` and `Recipe`:
+![RecipeClassDiagram.png](img/ClassDiagrams/recipesClassDiagram.png)
+
+
+### Ingredient Class
+The ingredient class represents a singular ingredient added by the user or loaded from the storage database.
+A valid ingredient object contains the following field:
++ Name
+
+The formatting of how an ingredient is displayed and saved is handled by this class.
+
+### IngredientList Class
+The IngredientList class represents a list of any non-negative integer number of ingredients.
+This class handles the addition, deletion and editing of ingredients.
+
+Shown below is a class diagram detailing the interaction between `IngredientList` and `Ingredient`:
+![ingredientClasses.png](img/ClassDiagrams/ingredientsClassDiagram.png)
+
+
+### Command Classes
+The command class deals with all the possible commands accepted by YMFC, with each command representing a specific 
+recognised user input (E.g. ListCommand representing the command to list all the recipes in the recipeList).
 
+#### 1. The `Command` base class
```

Comment:
I find this section a bit difficult to read as there's too much information squished together, could consider simplifying the class diagram or spacing out the sections